build.gradle

```diff
@@ -314,6 +315,7 @@ project('spring-integration-core') {
 		}
 		compile("io.fastjson:boon:$boonVersion", optional)
 		compile("com.esotericsoftware:kryo-shaded:$kryoShadedVersion", optional)
+		compile("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:$jsr310Version")
```

Comment:
I don't mind to test if `findAndRegisterModules()` works well, but I really mind to have this dependency as `compile`.
`testCompile` is fully enough for this task.

Thanks